src/node.rs

```diff
@@ -465,29 +468,110 @@ impl PbftNode {
     /// Panics if `finalize_block` fails. This is necessary because it means the validator wasn't
     /// able to publish the new block.
     pub fn try_publish(&mut self, state: &mut PbftState) -> Result<(), PbftError> {
-        // Try to finalize a block
-        if state.is_primary() && state.phase == PbftPhase::NotStarted {
-            debug!("{}: Summarizing block", state);
-            if let Err(e) = self.service.summarize_block() {
-                debug!(
+        // Only the primary takes care of this
+        if !state.is_primary() {
+            return Ok(());
+        }
+
+        if state.phase != PbftPhase::NotStarted {
+            return Err(PbftError::InternalError(format!(
+                "Node.try_publish requires phase PbftPhase::NotStarted, got {:?}!",
+                state.phase
+            )));
+        }
+
+        info!("{}: Summarizing block", state);
+
+        let summary = match self.service.summarize_block() {
+            Ok(bytes) => bytes,
+            Err(e) => {
+                warn!(
                     "{}: Couldn't summarize, so not finalizing: {}",
                     state,
                     e.description().to_string()
```

Comment:
This condition (where `summarize_block` returns an error) should still log with `debug` and should not return an error. This is not a true error, it's just a check; it just means the block isn't ready to be finalized yet so we need to wait longer.